### PR TITLE
UI: Added Hero Component and Course Detail Page

### LINKED ISSUE #3: Create Hero for Course Detail Page
Part 1 of #2

image

Compoenents are required to be made with material-ui: https://material-ui.com/

### PR DESCRIPTION
**Pull request by rizwan2000rm**
**Issue Number**

fixes #3

**Describe the changes you've made**

Added responsive Hero component for the Course Detail page. Also added a new CourseDetail page on route /course/:id

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -18,6 +18,7 @@
   },
   "plugins": ["react"],
   "rules": {
+    "react/react-in-jsx-scope": "off",
```
Comment (faisaljamil25) [inline .eslintrc.json:21]: I don't think react/react-in-jsx-scope should be disabled. All the errors related to this have been taken care of in the latest commits. What do you think @Siddhant-K-code ?

Comment (rizwan2000rm) [reply .eslintrc.json:21]: @faisaljamil25 I think we don't even need react-scope-in-jsx rule while linting in react v17.

Please read the latest new JSX transform documentation mentions the following
"With the new transform, you can use JSX without importing React"
https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html

Reverting and added #75

**Comment by rizwan2000rm** [comment]
@Abhishek-kumar09 @KeenWarrior Please review this PR, resolved all merge conflicts.

File: .eslintrc.json (hunk 1 above)
Comment (faisaljamil25) [reply .eslintrc.json]: You are correct, we can use JSX without importing React in v17. Thanks for the information.

**Pull request merged**